examples/pygfx-example.py

```diff
@@ -0,0 +1,209 @@
+        datefmt="[%X]",
+        handlers=[RichHandler(rich_tracebacks=True)],
+    )
+except ModuleNotFoundError:
+    pass
+
+
+logger = logging.getLogger(__name__)
+
+
+class SelectableObjectsCanvas(WgpuCanvas):
+    def handle_event(self, event):
+        if event["event_type"] == "pointer_down":
+            info = renderer.get_pick_info((event["x"], event["y"]))
+            wobject = info["world_object"]
+            if wobject:
+                if handle_event := getattr(wobject, "handle_event", None):
+                    handle_event({"event_type": "click"})
+
+        super().handle_event(event)
+
+
+sphere_geom = gfx.sphere_geometry(radius=0.5)
+materials = {
+    False: gfx.MeshPhongMaterial(color=[1, 1, 1]),
+    True: gfx.MeshPhongMaterial(color=[1, 0, 0]),
+}
+
+
+def PointCloud(props):
+    random.seed(0)
+
+    # Set some default values for props
+    props.setdefault("selected", -1)
+    props.setdefault("count", 50)
+
+    def set_selected(index):
+        logger.debug(f"select: {index}")
+        if props["selected"] == index:
+            props["selected"] = -1
+        else:
+            props["selected"] = index
+
+    def random_point(index, selected):
+        return h(
+            "Mesh",
+            {
+                "geometry": sphere_geom,
+                "material": materials[index == selected],
+                "position": [
+                    random.randint(-10, 10),
+                    random.randint(-10, 10),
+                    random.randint(-10, 10),
+                ],
+                "key": index,
+                "onClick": lambda event: set_selected(index),
+            },
+        )
+
+    selected = props["selected"]
+    number_of_points = props["count"]
+
+    return h(
+        "Group",
+        props,
+        *[random_point(i, selected) for i in range(number_of_points)],
+    )
+
+
+def Landmark(props):
+    props.setdefault("selected", False)
+
+    def toggle():
+        logger.debug("toggle")
+        props["selected"] = not props["selected"]
+
+    return h(
+        "Group",
+        props,
+        h(
+            "Point",
+            {
+                "scale": [2 if props["selected"] else 1] * 3,
+                "color": [1, 0.5, 0, 0.1],
+            },
+        ),
+        h(
+            "Point",
+            {
+                "scale": [0.5, 0.5, 0.5],
+                "color": [1, 1, 1, 1],
+                "onClick": lambda event: toggle(),
+            },
+        ),
+    )
+
+
+if __name__ == "__main__":
+    canvas = SelectableObjectsCanvas(size=(600, 400))
+    renderer = gfx.renderers.WgpuRenderer(canvas)
+    camera = gfx.PerspectiveCamera(70, 16 / 9)
+    camera.position.z = 15
+
+    controls = gfx.OrbitControls(camera.position.clone())
+    controls.add_default_event_handlers(canvas, camera)
+
+    gui = PyGui(renderer=PygfxRenderer(), event_loop_type=EventLoopType.QT)
+
+    # Should be possible to create this element
+    # by rendering JSX to dict...
```

Comment:
Jinja supports rendering templates to Python objects, so maybe we can use that out of the box when we get to it :)